### PR TITLE
Fix/sql backend destination

### PR DESCRIPTION
**Pull request by IvanHdzC**
fix for https://github.com/telefonicaid/fiware-cygnus/issues/2004

You can find test results on the attached files.

[agent.txt](https://github.com/telefonicaid/fiware-cygnus/files/6205591/agent.txt)
[cygnus.txt](https://github.com/telefonicaid/fiware-cygnus/files/6205592/cygnus.txt)

### CODE REVIEW

**Comment by AlvaroVega** [comment]
A CNR update is needed

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java [hunk 1]
```diff
@@ -34,37 +34,38 @@
 
     /**
      * Creates a table, given its name, if not exists in the given database.
-     * @param destination
+     * @param dataBase
      * @param tableName
```
Comment (mrutid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java:38]: missing param schema?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java [hunk 2]
```diff
@@ -34,37 +34,38 @@
...
      * Caps records from the given table within the given database according to the given maximum number.
-     * @param destination
+     * @param dataBase
      * @param tableName
      * @param maxRecords
      * @throws com.telefonica.iot.cygnus.errors.CygnusRuntimeError
      * @throws com.telefonica.iot.cygnus.errors.CygnusPersistenceError
      */
-    void capRecords(String destination, String tableName, long maxRecords) throws CygnusRuntimeError, CygnusPersistenceError;
+    void capRecords(String dataBase, String tableName, long maxRecords) throws CygnusRuntimeError, CygnusPersistenceError;
```
Comment (mrutid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java:68]: schema?

**Comment by AlvaroVega** [comment]
Still duplicated sentence: 
https://github.com/telefonicaid/fiware-cygnus/pull/2007/files#diff-f3724439dccac4f714857802a1a2c17bd80ae9e7822713b7fcb072de95cadb51R367

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java (hunk 1 above)
Comment (IvanHdzC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java:39]: Fixed in [f30b89d](https://github.com/telefonicaid/fiware-cygnus/pull/2007/commits/f30b89d404aac3d99c53fd399b32d9e176fe3589)

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java (hunk 2 above)
Comment (IvanHdzC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLBackend.java:69]: This method is only implemented by MySQL sink, That's why I didn't add the schema param. Do you thing It should be included?

**Comment by IvanHdzC** [comment]
> Still duplicated sentence:
> https://github.com/telefonicaid/fiware-cygnus/pull/2007/files#diff-f3724439dccac4f714857802a1a2c17bd80ae9e7822713b7fcb072de95cadb51R367

Fixed in [fc4f58a](https://github.com/telefonicaid/fiware-cygnus/pull/2007/commits/fc4f58abb42c38963cf4fc01b5e2354aba2cb3ab)

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/Enum.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.telefonica.iot.cygnus.backends.sql;
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/Enum.java:1]: License header in all new files, pls :)

Comment (IvanHdzC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/Enum.java:19]: Fixed in [8ef7b73](https://github.com/telefonicaid/fiware-cygnus/pull/2007/commits/8ef7b7341de12b4112270de5291166d7e6a49b80)

File: CHANGES_NEXT_RELEASE [hunk 4]
```diff
@@ -1 +1,4 @@
+[cygnus-ngsi] [PostgisSink] Fix PostgreSQL encoding when enabled.
+[cygnus-ngsi] [MysqlSink, PostgisSink, PosgreSQLSink] Update sinks to initialize persistance backend objects without database. (#2004)
+[cygnus-common] Remove database initialization on constructors. (#2004)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:3]: This PR has no any impact in documentation?

Just to check...

File: cygnus-ngsi-ld/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgisSink.java [hunk 5]
```diff
@@ -268,9 +269,7 @@ public void stop() {
     @Override
     public void start() {
         try {
-            if (buildDBName(null) != null) {
-                createPersistenceBackend(postgisHost, postgisPort, postgisUsername, postgisPassword, maxPoolSize, buildDBName(null), postgisOptions);
-            }
+            createPersistenceBackend(postgisHost, postgisPort, postgisUsername, postgisPassword, maxPoolSize, postgisOptions);
```
Comment (AlvaroVega) [inline cygnus-ngsi-ld/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgisSink.java:272]: Note that ngsi-ld sinks like this are not fully aligned with ngsiv2 sinks and maybe are not received latest features, bugfixes and refactors.

File: CHANGES_NEXT_RELEASE (hunk 4 above)
Comment (AlvaroVega) [reply CHANGES_NEXT_RELEASE:3]: It's just a bugfix, do not introduce a new behavior

**Review by AlvaroVega — APPROVED** [review]
LGTM

File: CHANGES_NEXT_RELEASE (hunk 4 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE:3]: NTC

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
@anmunoz this PR fixes some sinks in cygnus-ngsi that also are being used in cygnus-ngsi-ld. I'm telling you, just in case you want also to apply similar fixes to them.

**Pull request merged**